src/protocol/messages/delete_records.rs

```diff
@@ -0,0 +1,294 @@
+use std::io::{Read, Write};
+
+use crate::protocol::{
+    api_key::ApiKey,
+    api_version::{ApiVersion, ApiVersionRange},
+    error::Error,
+    messages::{
+        read_compact_versioned_array, read_versioned_array, write_compact_versioned_array,
+        write_versioned_array,
+    },
+    primitives::{CompactString, CompactStringRef, Int16, Int32, Int64, String_, TaggedFields},
+    traits::{ReadType, WriteType},
+};
+
+use super::{
+    ReadVersionedError, ReadVersionedType, RequestBody, WriteVersionedError, WriteVersionedType,
+};
+
+#[derive(Debug)]
+pub struct DeletePartitionRequest {
```

Comment:
Perhaps something like `DeleteRecordsRequestPartition` might make it clear this is for the `DeleteRecords` API